ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +992,264 @@ def test_server_option_with_unreachable_ad(self):
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_idoverride_with_auto_private_group_false(self):
+        """
+        override ad trusted user in default trust view
+        and set auto-private-groups=false to check the
+        output displayed
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        rangename = self.ad_domain.upper() + "_id_range"
+        self.master.run_command(
+            [
+                "ipa",
+                "idoverrideuser-add",
+                "Default Trust View",
+                "--uid=99999999",
+                "--gid=99999999",
+                nonposixuser,
+            ]
+        )
+        self.mod_idrange_auto_private_group(rangename, "false")
+        result = self.master.run_command(['id', nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        gid_reg = r"gid=(\d+)"
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_idoverride_with_auto_private_group_true(self):
+        """
+        override ad trusted user in default trust view
+        and set auto-private-groups=true to check the
+        output displayed
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        rangename = self.ad_domain.upper() + "_id_range"
+        self.master.run_command(
+            [
+                "ipa",
+                "idoverrideuser-mod",
+                "Default Trust View",
+                "--uid=88888888",
+                "--gid=88888888",
+                nonposixuser,
+            ]
+        )
+        self.mod_idrange_auto_private_group(rangename, "true")
+        result = self.master.run_command(['id', nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format
+            (nonposixuser)
+        )
+        gid_reg = r"gid=(\d+)"
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_idoverride_with_auto_private_group_hybrid(self):
+        """
+        override ad trusted user in default trust view
+        and set auto-private-groups=hybrid to check the
+        output displayed
+        """
```

Comment:
Please state in the comment what is the expected behavior (should the override win over the uid/gid values?)